.. INSaFLU documentation master file, created by
   sphinx-quickstart on Thu Jan 11 09:25:25 2018.
   You can adapt this file completely to your liking, but it should at least
   contain the root `toctree` directive.


***********************************
Welcome to INSaFLU's documentation!
***********************************

.. image:: _static/logo.png

**INSaFLU ("INSide the FLU") is an influenza-specific bioinformatics free web-based suite that deals with primary data (reads) towards the automatic generation of the output data that are actually the core first-line “genetic requests” for effective and timely influenza laboratory surveillance.**

https://insaflu-insa.min-saude.pt


Highlights
++++++++++

* open to all, free of charge, user-restricted accounts
* applicable to NGS data collected from any amplicon-based schema
* allows advanced, multi-step software intensive analyses in a user-friendly manner without previous training in bioinformatics
* automatic identification of influenza type and subtype/lineage, detection of putative mixed infections and intra-host minor variants
* allows integrating data in a cumulative manner, thus fitting the analytical dynamics underlying the continuous epidemiological surveillance during flu epidemics
* outputs are provided in nomenclature-stable and standardized formats and can be explored in situ or through multiple compatible downstream applications for fine-tune data analysis and visualization

How to cite
+++++++++++

If you use INSaFLU in your work, please cite **Borges V, Pinheiro M et al, 2018, bioRxiv**

Contact
+++++++
If you have any questions, comments or suggestions, please contact us: <vitor.borges@insa.min-saude.pt> or <j.paulo.gomes@insa.min-saude.pt>

*********
Contents:
*********

.. toctree::
   :maxdepth: 2

   guide_pre_ngs.rst
   uploading_data.rst
   project_creation.rst
   data_analysis.rst
   output_visualization.rst

